src/Service.Tests/SqlTests/RestApiTests/Insert/MsSqlInsertApiTests.cs

```diff
@@ -302,6 +314,58 @@ public async Task InsertOneInViewBadRequestTest()
             );
         }
 
+        /// <summary>
+        /// Test to validate that even when an insert DML trigger is enabled on a table, we still return the
+        /// latest data as it is present after the trigger gets executed. Whenever an insert DML trigger is enabled,
+        /// we use a subsequent SELECT query to get the data instead of using OUTPUT clause. To validate that the data is returned
+        /// as it is after the trigger is executed, we use the values which are updated by the trigger in the WHERE predicates of the verifying sql query.
```

Comment:
i don't remember if i asked for this pr, but would be helpful to describe why both autogen/non-autogen pk tests are needed. e.g. specific branching logic in mssqlquerybuilder that handles each case?